# HG changeset patch
# Parent 41b18bb0b9ad0497e7dff7625d95224ce550014e
Build: Disable stubdoms.

UPSTREAM: Needs to be made configurable

diff -r 41b18bb0b9ad Makefile
--- a/Makefile	Wed Feb 29 16:26:16 2012 +0000
+++ b/Makefile	Wed Feb 29 16:28:15 2012 +0000
@@ -15,16 +15,12 @@ include buildconfigs/Rules.mk
 
 # build and install everything into the standard system directories
 .PHONY: install
-install: install-xen install-kernels install-tools install-stubdom install-docs
+install: install-xen install-kernels install-tools install-docs
 
 .PHONY: build
 build: kernels
 	$(MAKE) -C xen build
 	$(MAKE) -C tools build
-	$(MAKE) -C stubdom build
-ifeq (x86_64,$(XEN_TARGET_ARCH))
-	XEN_TARGET_ARCH=x86_32 $(MAKE) -C stubdom pv-grub
-endif
 	$(MAKE) -C docs build
 
 # The test target is for unit tests that can run without an installation.  Of
@@ -37,7 +33,7 @@ test:
 # build and install everything into local dist directory
 .PHONY: dist
 dist: DESTDIR=$(DISTDIR)/install
-dist: dist-xen dist-kernels dist-tools dist-stubdom dist-docs dist-misc
+dist: dist-xen dist-kernels dist-tools dist-docs dist-misc
 
 dist-misc:
 	$(INSTALL_DIR) $(DISTDIR)/check
@@ -50,11 +46,10 @@ dist-%: install-%
 	@: # do nothing
 
 # Legacy dist targets
-.PHONY: xen tools stubdom kernels docs
+.PHONY: xen tools kernels docs
 xen: dist-xen
 tools: dist-tools
 kernels: dist-kernels
-stubdom: dist-stubdom
 docs: dist-docs
 
 .PHONY: prep-kernels
@@ -77,13 +72,6 @@ endif
 install-kernels:
 	for i in $(XKERNELS) ; do $(MAKE) $$i-install || exit 1; done
 
-.PHONY: install-stubdom
-install-stubdom: tools/ioemu-dir install-tools
-	$(MAKE) -C stubdom install
-ifeq (x86_64,$(XEN_TARGET_ARCH))
-	XEN_TARGET_ARCH=x86_32 $(MAKE) -C stubdom install-grub
-endif
-
 tools/ioemu-dir:
 	$(MAKE) -C tools ioemu-dir-find
 
@@ -127,10 +115,6 @@ world:
 clean::
 	$(MAKE) -C xen clean
 	$(MAKE) -C tools clean
-	$(MAKE) -C stubdom crossclean
-ifeq (x86_64,$(XEN_TARGET_ARCH))
-	XEN_TARGET_ARCH=x86_32 $(MAKE) -C stubdom crossclean
-endif
 	$(MAKE) -C docs clean
 
 # clean, but blow away kernel build tree plus tarballs
@@ -138,10 +122,6 @@ endif
 distclean:
 	$(MAKE) -C xen distclean
 	$(MAKE) -C tools distclean
-	$(MAKE) -C stubdom distclean
-ifeq (x86_64,$(XEN_TARGET_ARCH))
-	XEN_TARGET_ARCH=x86_32 $(MAKE) -C stubdom distclean
-endif
 	$(MAKE) -C docs distclean
 	rm -rf dist patches/tmp
 	for i in $(ALLKERNELS) ; do $(MAKE) $$i-delete ; done
@@ -165,7 +145,6 @@ help:
 	@echo '  install-xen      - build and install the Xen hypervisor'
 	@echo '  install-tools    - build and install the control tools'
 	@echo '  install-kernels  - build and install guest kernels'
-	@echo '  install-stubdom  - build and install the stubdomain images'
 	@echo '  install-docs     - build and install user documentation'
 	@echo ''
 	@echo 'Building targets:'
@@ -174,7 +153,6 @@ help:
 	@echo '                     trees then make dist'
 	@echo '  xen              - build and install Xen hypervisor'
 	@echo '  tools            - build and install tools'
-	@echo '  stubdom          - build and install the stubdomain images'
 	@echo '  kernels          - build and install guest kernels'
 	@echo '  kbuild           - synonym for make kernels'
 	@echo '  docs             - build and install user documentation'
